File: php/inorderTraversal.php
<?php

/**
 * Definition for a binary tree node.
 * class TreeNode {
 *     public $val = null;
 *     public $left = null;
 *     public $right = null;
 *     function __construct($val = 0, $left = null, $right = null) {
 *         $this->val = $val;
 *         $this->left = $left;
 *         $this->right = $right;
 *     }
 * }
 */

class TreeNode
{
    public $val = null;
    public $left = null;
    public $right = null;

    function __construct($val = 0, $left = null, $right = null)
    {
        $this->val = $val;
        $this->left = $left;
        $this->right = $right;
    }
}

class Solution {

    /**
     * @param TreeNode $root
     * @return Integer[]
     */
    function inorderTraversal($root) {
        $result = [];

        if (is_null($root)) {
            return $result;
        }

        if (!is_null($root->left)) {
            $result = array_merge($result, $this->inorderTraversal($root->left));
        }
        $result[] = $root->val;
        if (!is_null($root->right)) {
            $result = array_merge($result, $this->inorderTraversal($root->right));
        }
        return $result;
    }
}

var_dump(
    (new Solution())->inorderTraversal(
        (new TreeNode(
            1,
            null,
            (new TreeNode(
                2,
                (new TreeNode(
                    3, null, null
                )),
                null
            ))
        ))
    )
);

File: go/inorderTraversal.go
package main

type TreeNode struct {
	Val int
	Left *TreeNode
	Right *TreeNode
}

/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */
 func inorderTraversal(root *TreeNode) []int {
    var result []int
    var stack [] *TreeNode

    for ;; {
        if root == nil {
            stackLen := len(stack)
            if stackLen > 0 {
                root = stack[stackLen - 1]
                stack = stack[:stackLen - 1]
            }
        }

        if root != nil {
            if root.Left != nil {
				stack = append(stack, root)
				leftNode := root.Left
				root.Left = nil
                root = leftNode
            } else {
				result = append(result, root.Val)
				rightNode := root.Right
				root = nil
				if rightNode != nil {
					root = rightNode
				}
			}
        } else {
            break
        }
    }

    return result
}
